src/NServiceBus.Core/Causation/ConversationId.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace NServiceBus
+{
+    using System;
+
+    /// <summary>
+    /// Holds the conversation ID to use for the outgoing message.
+    /// </summary>
+    public struct ConversationId
```

Comment:
One of the reasons I questioned the introduction of this struct is that it doesn't offer the protection you're hoping for.

All structs have an implicit parameterless constructor that initializes all fields to their default values, so even with this struct, users can just `return new ConversationId();` from the func they give us, and we'll still have a null `Value` that completely skips the validation.

Making this a class doesn't solve it either, since they could then just `return null;`, and we'd also be adding a new allocation for the `Default` strategy as well.